driver/pace/driver/run.py

```diff
@@ -116,82 +130,179 @@ class SerialboxConfig(InitializationConfig):
     """
 
     path: str
-    npx: int
-    npy: int
-    npz: int
-    layout: List[int]
+    serialized_grid: bool
     backend: str
+    rebuild: bool
+    validate_args: bool
+    format_source: bool
+    device_sync: bool
 
     @property
     def start_time(self) -> datetime:
         return datetime(2000, 1, 1)
 
+    @property
+    def f90_namelist(self) -> f90nml.Namelist:
+        return f90nml.read(self.path + "/input.nml")
+
+    @property
+    def namelist(self) -> Namelist:
+        return Namelist.from_f90nml(self.f90_namelist)
+
+    @property
+    def stencil_config(self) -> pace.dsl.stencil.StencilConfig:
+        return pace.dsl.stencil.StencilConfig(
+            backend=self.backend,
+            rebuild=self.rebuild,
+            validate_args=self.validate_args,
+        )
+
+    def get_serialized_grid_damping_coeff_and_driver_grid(
+        self, communicator: pace.util.CubedSphereCommunicator
+    ) -> pace.stencils.testing.grid.Grid:
+        ser = self.serializer(communicator)
+        grid_savepoint = ser.get_savepoint("Grid-Info")[0]
+        grid_data = {}
+        grid_fields = ser.fields_at_savepoint(grid_savepoint)
+        for field in grid_fields:
+            grid_data[field] = ser.read(field, grid_savepoint)
+            if len(grid_data[field].flatten()) == 1:
+                grid_data[field] = grid_data[field][0]
+        savepoint_in = ser.get_savepoint("FVDynamics-In")[0]
+        for field in ["ak", "bk", "ptop"]:
+            grid_data[field] = ser.read(field, savepoint_in)
+            if len(grid_data[field].flatten()) == 1:
+                grid_data[field] = grid_data[field][0]
+        grid = TranslateGrid(
+            grid_data, communicator.rank, self.namelist.layout, backend=self.backend
+        ).python_grid()
+        grid.grid_data.ak = grid_data["ak"]
+        grid.grid_data.bk = grid_data["bk"]
+        grid.grid_data.ptop = grid_data["ptop"]
+        damping_coefficients = DampingCoefficients(
+            divg_u=grid_data["divg_u"],
+            divg_v=grid_data["divg_v"],
+            del6_u=grid_data["del6_u"],
+            del6_v=grid_data["del6_v"],
+            da_min=grid_data["da_min"],
+            da_min_c=grid_data["da_min_c"],
+        )
+        stencil_factory = StencilFactory(
+            config=self.stencil_config, grid_indexing=grid.grid_indexing
+        )
+        driver_grid_info_object = TranslateUpdateDWindsPhys(
+            grid, self.namelist, stencil_factory
+        )
+        extra_grid_data = driver_grid_info_object.collect_input_data(
+            ser, ser.get_savepoint("FVUpdatePhys-In")[0]
+        )
```

Comment:
It's unfortunate that we need a stencil factory to do something that involves creating no stencils, but it looks like that is in fact the case because the translate infrastructure uses low-level configuration from StencilFactory in combination with `make_storage_from_shape` instead of using `QuantityFactory`.

I think though that we only need the `backend` value. You should be able to remove the other configuration options above (rebuild, validate_args, format_source, device_sync) and provide a nonsensical value for them (None, probably) that would cause errors if we ever did create stencils through this pathway. This would reduce the dataclass API above to only what's needed.

Reply:
Sounds good, keeping `backend` only here. Again, this should look cleaner when we consolidate `GridData` and `DriverGridData`.

Reply:
Making a jira card for that now